src/meshes/t8code_mesh.jl

```diff
@@ -100,6 +100,125 @@ function Base.show(io::IO, ::MIME"text/plain", mesh::T8codeMesh)
     end
 end
 
+"""
+    T8codeMesh{NDIMS, RealT}(forest, boundary_names; polydeg = 1, mapping=identity)
+
+Create a 'T8codeMesh'.
+
+# Arguments
+- `forest`: Pointer to a committed forest.
+- `boundary_names`: List of boundary names.
+- `polydeg::Integer`: Polynomial degree used to store the geometry of the mesh.
+                      The mapping will be approximated by an interpolation polynomial
+                      of the specified degree for each tree.
+- `mapping`: a function of `NDIMS` variables to describe the mapping that transforms
+             the imported mesh to the physical domain. Use `nothing` for the identity map.
+"""
+function T8codeMesh{NDIMS, RealT}(forest::Ptr{t8_forest}, boundary_names; polydeg = 1,
+                                  mapping = nothing) where {NDIMS, RealT}
+    # In t8code reference space is [0,1].
+    basis = LobattoLegendreBasis(RealT, polydeg)
+    nodes = 0.5 .* (basis.nodes .+ 1.0)
+
+    cmesh = t8_forest_get_cmesh(forest)
+    num_trees = t8_forest_get_num_global_trees(forest)
+
+    tree_node_coordinates = Array{RealT, NDIMS + 2}(undef, NDIMS,
+                                                    ntuple(_ -> length(nodes), NDIMS)...,
+                                                    num_trees)
+
+    coords_ref = Vector{Float64}(undef, 3)
+
+    # Calculate node coordinates of reference mesh.
+    if NDIMS == 2
+        num_corners = 4 # quadrilateral
+
+        # Testing for negative element volumes.
+        verts = zeros(3, num_corners)
+        for itree in 1:num_trees
+            veptr = t8_cmesh_get_tree_vertices(cmesh, itree - 1)
+
+            # Note, `verts = unsafe_wrap(Array, veptr, (3, 1 << NDIMS))`
+            # sometimes does not work since `veptr` is not necessarily properly
+            # aligned to 8 bytes.
+            for icorner in 1:num_corners
+                verts[1, icorner] = unsafe_load(veptr, (icorner - 1) * 3 + 1)
+                verts[2, icorner] = unsafe_load(veptr, (icorner - 1) * 3 + 2)
+            end
+
+            # Check if tree's node ordering is right-handed or print a warning.
+            let z = zero(eltype(verts)), o = one(eltype(verts))
+                u = verts[:, 2] - verts[:, 1]
+                v = verts[:, 3] - verts[:, 1]
+                w = [z, z, o]
+
+                # Triple product gives signed volume of spanned parallelepiped.
+                vol = dot(cross(u, v), w)
+
+                if vol < z
+                    @warn "Discovered negative volumes in `cmesh`: vol = $vol"
```

Comment:
```suggestion
                    error("Discovered negative volumes in `cmesh`: vol = $vol")
```
Should this really continue if the volume is negative? I mean, is there a use case where it makes sense to proceed even if the ordering is not right-handed?

If not, I propose to upgrade this to a warning (same thing below)